File: Java/Tema4_Ejercicio3.java
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 3
 * ===============================================
 * Escriba un programa que imprima un rectángulo usando bucles anidados.
 */

import java.util.Scanner;

public class Tema4_Ejercicio3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite el ancho del rectángulo: ");
        int ancho = scanner.nextInt();
        
        System.out.print("Digite el alto del rectángulo: ");
        int alto = scanner.nextInt();
        
        System.out.println("\n=== RECTÁNGULO ===");
        
        for (int i = 0; i < alto; i++) {
            for (int j = 0; j < ancho; j++) {
                System.out.print("*");
            }
            System.out.println();
        }
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite el ancho del rectángulo: 5
 * Digite el alto del rectángulo: 3
 * 
 * === RECTÁNGULO ===
 * *****
 * *****
 * *****
 */

File: Python/Tema4_Ejercicio3.py
"""
TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 3
==============================================
Bucles anidados y patrones.

EXPLICACIÓN:
- Bucles dentro de bucles
- Patrones geométricos
- Matrices simples
"""

print("=== TABLA DE MULTIPLICACIÓN ===\n")

for i in range(1, 4):
    for j in range(1, 4):
        print(f"{i*j:3}", end=" ")
    print()

print("\n=== RECTÁNGULO DE ASTERISCOS ===\n")

ancho = 5
alto = 3

for i in range(alto):
    for j in range(ancho):
        print("*", end="")
    print()

print("\n=== TRIÁNGULO ASCENDENTE ===\n")

for i in range(1, 6):
    for j in range(i):
        print("*", end="")
    print()

print("\n=== TRIÁNGULO DESCENDENTE ===\n")

for i in range(5, 0, -1):
    for j in range(i):
        print("*", end="")
    print()

"""
SALIDA ESPERADA:
=== TABLA DE MULTIPLICACIÓN ===

  1   2   3
  2   4   6
  3   6   9

=== RECTÁNGULO DE ASTERISCOS ===

*****
*****
*****

=== TRIÁNGULO ASCENDENTE ===

*
**
***
****
*****

=== TRIÁNGULO DESCENDENTE ===

*****
****
***
**
*
"""
